### DISCUSSION TITLE
Z-Wave JS UI stopped working after upgrade to 13.7.0

### DISCUSSION THREAD

**Discussion by genterminl** [zwave-js/zwave-js#7414] [Request Support / Investigate Issue] [answered]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

Originally posted at
https://community.home-assistant.io/t/problem-with-z-wave-js-ui-upgrade-from-13-6-1-to-13-7-0/794159/4
This report has been assembled over many days with multiple restarts, repeating tests, and searching logs. 

The system (HAOS, running in VBox on a Gentoo Linux host) was working fine for months; problem first noticed after Z-Wave JS UI updated from 3 16.1 to 3.17.0.  The upgrade took more than one attempt, with restarts of the host.  Once the upgrade was complete, all my devices were grayed out in the overview, clicking Z-Wave JS in the sidebar says
`The add-on seems not to be ready, it might still be starting. So you want to try again?`
Retry gives the same error, no retry says `502.  Bad gateway`.  Settings/Add-ons shows Z-Wave JS is installed but not running, Z-Wave JS UI looks OK, but "Open Web UI" gives same result as above.

Restoring the backup from prior to the upgrade restores all functionality  (tested more than once.)

The ZWave JS UI log shows
`s6-rc-compile: fatal: invalid /etc/s6-overlay/s6-rc.d/discovery/type: must be oneshot, longrun, or bundle`
I thought I had seen more of that message, but couldn't find any more at the time.  This is why it was suggested I post here, although I have NOT been able to figure out which service is actually generating that error.  Since then, the JS UI log includes various repeats of
```
s6-rc-compile: fatal: invalid /etc/s6-overlay/s6-rc.d/discovery/type: must be oneshot, longrun, or bundle
s6-rc: fatal: unable to take locks: No such file or directory
s6-linux-init-shutdownd: warning: /run/s6/basedir/scripts/rc.shutdown exited 111
```
although the latest timestamp in the log (above all of these errors) is from five days ago, despite multiple restarts of the JS UI and even the whole VM.  One more thing I've found is an error on the Settings/Devices and Services/Integrations/Z-Wave page saying
`Failed setup, will retry: Failed to connect: Cannot connect to host a0d7b954-zwavejs2mqtt:3000 ssl:default [Connect call failed (‘172.30.33.0’, 3000)]`
but the config shows URL ws://a0d7b954-zwavejs2mqtt:3000 which seems to be what the docs call for.  Poking around some more shows the hub as “unavailable.”  

I can't tell if I might have more than one actual problem, which seems unlikely, but I don't know which error is important and which are just red herrings.  The hardware is all OK, because downgrading Z-Wave JS UI restores all functionality.

I'll be glad to post more logs (full or extract) as long as I can find them.

### Software versions

Driver (node-zwave-js): 13.10.3
Z-Wave JS UI: 13.7.0
Home Assistant Z-Wave Integration: ...(where do I find this?)
Home Assistant Z-Wave JS Addon: 0.9.0
ioBroker.zwave2 Adapter: ...(where do I find this?)

Mosquitto broker: 6.4.1

I don't think I have anything non-standard.

### Z-Wave Controller (Stick/Dongle/...)

Hub is a Zooz 800 ZST39 LR, which is definitely connected, showing up in lsusb on both the host and in the VM, and it is one of the 7 Z_Wave devices shown.  Firmware 1.20

### Device information

I don't think this is device related, as downgrading Z-Wave JS UI completely restored functionality.

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [ ] The log includes a re-interview of the problematic device _(if applicable)_.

- [ ] The log includes the problematic interaction with the device _(if applicable)_.

- [ ] I provided the node ID of the problematic device _(if applicable)_.

### Upload Logfile

[zwave_2024-11-17.log](https://github.com/user-attachments/files/17792669/zwave_2024-11-17.log)

[a comment by an automated account is not shown]

**Reply by genterminl** [reply]
None of the examples of how to generate the Driver Log match my system, but this file matches what one doc says it should look like (content, not name.)  Also, the log implies the driver version is 13.10.0, but I see no way to upgrade to apparently available newer versions - unless I just need to wait for  HomeAssistant to make it available.

**Comment by AlCalzone** [comment]
The driver logs look completely normal to me. I don't know how to get to Z-Wave JS UI's application logs in this situation to see if the error lies there.

**Reply by genterminl** [reply]
Thanks for looking.  I've attached both the JS and JS UI logs.  What I absolutely don't understand about the JS UI log is that it represents multiple restarts of the addon, HAOS, and the entire VM.  I have no idea why all those restarts just add lines to the log, but nothing timestamped for about a week.  
[a0d7b954_zwavejs2mqtt_2024-11-20T02-26-16.101Z.log](https://github.com/user-attachments/files/17823863/a0d7b954_zwavejs2mqtt_2024-11-20T02-26-16.101Z.log)
[core_zwave_js_2024-11-20T02-26-48.282Z.log](https://github.com/user-attachments/files/17823864/core_zwave_js_2024-11-20T02-26-48.282Z.log)

**Reply by genterminl** [reply]
Also note that at the top of the page displaying the JS log, it says "Failed to get core_zwave_js logs, Error in input stream" so I'm not sure what it actually downloaded for me to post above.

**Reply by AlCalzone** [reply]
Uh are you running both the Z-Wave JS and the Z-Wave JS UI addon? That doesn't work, they will be fighting over the serial port.

**Reply by genterminl** [reply]
Both addons are installed, but definitely only one is running at a time.  As part of my testing, I did stop JS UI and start JS addon, and at least one of my devices (switch) showed up and worked.  Remember, this problem just started with the recent upgrade of JS UI.  Before that, everything was working fine for months, including other upgrades.

**Reply by AlCalzone** [reply, marked as answer]
I see. The stuff regarding `s6-` seems to be specific to the [HA addon](https://github.com/hassio-addons/addon-zwave-js-ui).
I have no experience with how that works internally. The error message complains about `.../discovery/type` being wrong, but https://github.com/hassio-addons/addon-zwave-js-ui/blob/main/zwave-js-ui/rootfs/etc/s6-overlay/s6-rc.d/discovery/type seems to contain one of the valid values.

So maybe your pulled image is broken locally? No clue how to fix that though.

**Comment by genterminl** [comment]
It seems you did end up with the right conclusion, and thanks to a suggestion from freshcoast in the HA community forum, I stopped the ZUI addon, removed and re-downloaded the docker image for it, restarted the addon, and everything started working again.  So - the problem was indeed with the JS UI Addon - but apparently not in the released version, only because my copy got corrupted during download or installation.

Thanks for all your support @AlCalzone.  (should I mark this or your previous comment as the solution?)
